src/test/java/org/springframework/data/couchbase/transactions/CouchbaseReactiveTransactionsWrapperIntegrationTests.java

```diff
@@ -157,8 +159,43 @@ public void wrapperReplaceWithCasConflictResolvedViaRetryReactive() {
 		//}
 	}
 
+	@DisplayName("Forcing CAS mismatch causes a transaction retry")
+	@Test
+	public void casMismatchCausesRetry() {
+		UUID id = UUID.randomUUID();
+		Person person = new Person(id, "Walter", "White");
+		operations.insertById(Person.class).one(person);
+		AtomicInteger attempts = new AtomicInteger();
+
+		// Needs to take place in a separate thread to bypass the ThreadLocalStorage checks
+		Thread forceCASMismatch = new Thread(() -> {
+			Person fetched = operations.findById(Person.class).one(id.toString());
+			operations.replaceById(Person.class).one(fetched.withFirstName("Changed externally"));
+		});
```

Comment:
the couchbase 'cluster' is exposed through operations.getCouchbaseClientFactory().getCluster()

Reply:
Those are good points.   
Because spring inserts automatically add a _class property and spring queries automatically add a predicate _class = "<class-name>",  docs with non-spring insert/replace/update will not have the _class property, and spring queries will not find them.